tests/ETHTornado.test.ts

```diff
@@ -131,150 +173,169 @@ describe("ETHTornado", function () {
             pathIndices: path_index,
         };
 
-        const wasmPath = path.join(__dirname, "../build/withdraw.wasm");
+        const wasmPath = path.join(__dirname, "../build/withdraw_js/withdraw.wasm");
         const zkeyPath = path.join(__dirname, "../build/circuit_final.zkey");
 
-        const { proof } = await groth16.fullProve(witness, wasmPath, zkeyPath);
-        const solProof = parseProof(proof);
-
-        const txWithdraw = await tornado
-            .connect(relayerSigner)
-            .withdraw(solProof, root, nullifierHash, recipient, relayer, fee);
-        const receiptWithdraw = await txWithdraw.wait();
-        console.log("Withdraw gas cost", receiptWithdraw.gasUsed.toNumber());
-    }).timeout(500000);
-
-    it("prevent a user withdrawing twice", async function () {
-        const [userOldSigner, relayerSigner, userNewSigner] =
-            await ethers.getSigners();
-        const deposit = Deposit.new();
-        const tx = await tornado
-            .connect(userOldSigner)
-            .deposit(deposit.commitment, { value: ETH_AMOUNT });
-        const receipt = await tx.wait();
-        const events = await tornado.queryFilter(
-            tornado.filters.Deposit(),
-            receipt.blockHash
-        );
-        deposit.leafIndex = events[0].args.leafIndex;
-
-        const tree = new MerkleTree(HEIGHT, "test", new PoseidonHasher());
-        await tree.insert(deposit.commitment);
-
-        const nullifierHash = deposit.nullifierHash;
-        const recipient = await userNewSigner.getAddress();
-        const relayer = await relayerSigner.getAddress();
-        const fee = 0;
+        console.log("pre proof");
 
-        const { root, path_elements, path_index } = await tree.path(
-            deposit.leafIndex
-        );
+        // XXX Here - let's try using generated witness_calculator instead
+        const wc = require("../build/withdraw_js/witness_calculator");
 
-        const witness = {
-            // Public
-            root,
-            nullifierHash,
-            recipient,
-            relayer,
-            fee,
-            // Private
-            nullifier: BigNumber.from(deposit.nullifier).toBigInt(),
-            pathElements: path_elements,
-            pathIndices: path_index,
-        };
+        // Error in template MerkleTreeChecker_136 line: 40
```

Comment:
Here's the current error, seems to be pointing at this https://github.com/ChihChengLiang/poseidon-tornado/blob/0724975fdb42bbf520c839f3e9c66031a8dc4f8e/circuits/merkleTree.circom#L40

Using existing code path with full proof gives a generic assert error (see below). This is based on https://github.com/iden3/snarkjs/issues/126